Core.Collectors/IO/CloudQueueMsiWrapper.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.Core.Collectors.IO
+{
+    public class CloudQueueMsiWrapper : IQueue
+    {
+        private readonly CloudQueue queue;
+        private readonly string queueName;
+        private readonly string storageAccountNameEnvironmentVariable;
+
+        public CloudQueueMsiWrapper(CloudQueue queue, string storageAccountNameEnvironmentVariable)
+        {
+            this.queue = queue;
+            this.queueName = queue.Name;
+            this.storageAccountNameEnvironmentVariable = storageAccountNameEnvironmentVariable;
+        }
+
+        public Task PutObjectAsJsonStringAsync(object obj)
+        {
+            string message = JsonConvert.SerializeObject(obj);
+            return this.PutMessageAsync(message);
+        }
+
+        public async Task PutMessageAsync(string message)
+        {
+            CloudQueue queue = await AzureHelpers.GetStorageQueueUsingMsiAsync(queueName, storageAccountNameEnvironmentVariable).ConfigureAwait(false);
```

Comment:
Do we have to reconnect to the CloudQueue  every time we add a message? Could we do this in the constructor and reuse the object instead?

Reply:
CloudQueue can vary depending on where it is instantiated. Thus, making it have a token expiration that may vary with others, so it has to be called individually.